Задача 5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить
данную гипотезу, если известно, что размеры изделий подчинены нормальному закону
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [7]:
import numpy as np

In [8]:
M = 2.5
sample = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = 10
a = 0.05

Так как дисперсия неизвестна , используем t – критерий.
Проводим двусторонний тест.

In [9]:
mx = np.mean(sample)
t = (mx - M) / (np.std(sample, ddof=1) / np.sqrt(n))
print('t = ', t)
print('среднее значение по выборке =', mx)

t =  0.5630613661802959
среднее значение по выборке = 2.5279999999999996


Нулевая гипотеза утверждает, что партия изготавливается со средним арифметическим 2,5 см.
Альтернативная гипотеза утверждает, что партия изготавливается со средним арифметическим, отличным от 2,5 см.

для доверительной вероятности 97,5% и числа степеней свободы n - 1 = 9, табличное t = 2.262.
Т.к. среднее значение по выборке = 2.528 > 2.5, а t расчётное > t табличного (0.563 > 2.262) принимаем альтернативную гипотезу (партия изготавливается со средним арифметическим, отличным от 2,5 см)